### DISCUSSION TITLE
Shipping Discount API - Allow to discount single deliveryOption into FunctionResult deliveryGroups

### DISCUSSION THREAD

**Discussion by AndreaMorone** [Shopify/function-examples#400] [Feedback: Discounts APIs]
Hey there! Just wanted to drop a suggestion for the Shipping Discount functions API. It would be great if we could allow discounts on single delivery options within the FunctionResult, rather than discounting all the deliveries within the delivery groups. This would give us more flexibility and fine-grained control over applying discounts.

In the following example I wanted to discount only the **STANDARD** shipping rate by half price, but also **Express Shipping** is getting discounted.

![screenshot_delivery](https://github.com/Shopify/function-examples/assets/7823934/8d198e37-19ad-4589-a052-27baf4faaed6)

Would be great if we could change the output passing out to the function an array of deliveryGroups with an array of deliveryOptions to discount 👇🏻

**From This**

![Xnapper-2023-11-20-19 03 22](https://github.com/Shopify/function-examples/assets/7823934/f9a4363e-64d9-49bf-a4b5-2c78fd39108e)

**To This**

![Xnapper-2023-11-20-19 17 15](https://github.com/Shopify/function-examples/assets/7823934/fa301924-34e0-4628-b9e8-a9f628bc3ce8)

**Comment by AndreaMorone** [comment]
As per the new Shopify api version these seems now possible https://shopify.dev/docs/api/functions/reference/shipping-discounts/graphql/common-objects/target

**Reply by narviii** [reply]
Btw, how can I get this deliveryOptions handle to be able to change it in the user interface?

**Reply by AndreaMorone** [reply]
You cannot change the handle in the UI, maybe you mean the title? If so you need to use delivery customizations functions

**Reply by narviii** [reply]
I don't want to change the handle, I want to read what handles(delivery options) are available from admin API. Basically, I want for user to be able select from UI the option he wants to discount. In order to do so I need to know it's handle. How I can do that?

**Reply by thosakwe** [reply]
You can't read the handles via the admin API.

As far as I can tell, the handles only exist during checkout.

You'll have to match the delivery option based on its title and then read its handle from the function input.
